abel/rbasex.py

```diff
@@ -0,0 +1,751 @@
+            negative overshoots than ``'L2'``.
+        ``'SVD'``:
+            truncated SVD (singular value decomposition) with
+            N = `strength` × **rmax** largest singular values removed for each
+            angular order. This mimics the approach proposed in pBasex.
+        ``'pos'``:
+            non-parameterized method, finds the best (in the least-squares
+            sense) solution with non-negative :math:`\cos^n\theta \sin^m\theta`
+            terms (see :meth:`~abel.tools.vmi.Distributions.Results.cossin`).
+            For **order** = 0, 1, and 2 (with **odd** = `False`) this is
+            equivalent to :math:`I(r, \theta) \geqslant 0`; for higher orders
+            this assumption is stronger than :math:`I \geqslant 0` and
+            corresponds to no interference between different multiphoton
+            channels. Not implemented for odd orders > 1.
+
+            Notice that this method is nonlinear, which also means that it is
+            considerably slower than the linear methods and the transform
+            operator cannot be cached.
+
+        In all cases, `strength` = 0 provides no regularization. For the
+        Tikhonov methods, `strength` ~ 100 is a reasonable value for megapixel
+        images. For truncated SVD, `strength` must be < 1; `strength` ~ 0.1 is
+        a reasonable value; `strength` ~ 0.5 can produce noticeable ringing
+        artifacts.
+    out : str or None
+        shape of the output image:
+
+        ``'same'`` (default):
+            same shape and origin as the input
+        ``'fold'`` (fastest):
+            Q0 quadrant (for ``odd=False``) or right half (for ``odd=True``) up
+            to **rmax**, but limited to the largest input-image quadrant (or
+            half)
+        ``'unfold'``:
+            like ``'fold'``, but symmetrically “unfolded”
+        ``'full'``:
+            all pixels with radii up to **rmax**
+        ``'full-unique'``:
+            the unique part of ``'full'``: Q0 quadrant for ``odd=False``, right
+            half for ``odd=True``
+        ``None``:
+            no image (**recon** will be ``None``). Can be useful to avoid
+            unnecessary calculations when only the transformed radial
+            distributions (**distr**) are needed.
+    basis_dir : str, optional
+        path to the directory for saving / loading the basis set (useful only
+        for the inverse transform without regularization; time savings in other
+        cases are small and might be negated by the disk-access overhead).
+        If ``None``, the basis set will not be loaded from or saved to disk.
+    verbose : bool
+        print information about processing (for debugging)
+
+    Returns
+    -------
+    recon : 2D numpy array or None
+        the transformed image. Is centered and might have different dimensions
+        than the input image.
+    distr : Distributions.Results object
```

Comment:
This is fundamentally different from all of the other `_transform` functions, which only return the image array. I see why it needs to work this way, but it's potentially confusing.

Reply:
`distr` is encapsulates `(r)basex` results, much like the `Transform` class contains more than just the Abel transformed image. It is neat and tidy, and e.g. should be used for `linbasex` to contain the `Beta` arrays unique to it.

Reply:
Yes, this is very much like in `linbasex`, except that `rbasex` always returns 2 values instead of 1 or 3 depending on input parameters.